proposals/wg-image-compatibility.md

```diff
@@ -0,0 +1,106 @@
+the image compatibility specification, which is a key feature that describes the requirements of special containerized applications
+that have hard requirements for specific kernel versions, configurations, modules, or out-of-tree drivers.
+
+For example, a container application  free5GC UPF requiring the gtp5g kernel module  will only be compatible with kernel 5.0.0-23-generic 
+or 5.4.x due to the module's hard kernel version requirements - https://free5gc.org/guide/3-install-free5gc/.
+If SR-IOV or GPU is required by the container, other requirements have to be specified against the host OS.
+Kata Containers use cases are the best example: https://github.com/kata-containers/kata-containers/tree/main/docs/use-cases.
+
+
+The described incompatibility problem affects distributed systems.
+In multi-cloud, hybrid cloud, or even private cloud scenarios, one containerized cloud-native application will need to run
+on different Kubernetes distributions that use different host operating systems on different worker nodes.
+
+Operating systems can use different kernel versions with different configurations.
+Different operating system distributions may integrate different devices and their corresponding drivers,
+even for the same device different versions of drivers may be used. Therefore, different operating system distributions are not identical.
+Even within the same OS distro, different releases are also not identical.
+
+An image compatibility specification would help container image authors describe compatibility requirements in a standard way.
+The specification will be uploaded with the image to the image registry. This makes hard container compatibility requirements discoverable,
+programmable, and will support different consumers and cover use cases where the application requires a specific compatible environment.
+
+For further reference please follow:
+* [OCI Image Compatibility - the spec reasoning](https://docs.google.com/presentation/d/1F9GnCm2sULuyTJ5BEFZlL8Qjab81DK7g9Oy6qBfe5Qs/edit?usp=sharing)
+* [Revisit container and host OS compatibility](https://docs.google.com/document/d/1lzwh8DGMu5vXXHwJmnewYIMffkcOEvH8owX4UYjRcw0/edit#heading=h.fcxt9vheg92c)
+
+## Scope
+
+* Define image compatibility spec that describes special host OS requirements of containerized application.
+* The spec should describe container requirements for Linux, FreeBSD and Windows.
+* The final list of supported fields (like kernel modules, configuration, out-of-tree drivers etc.)
+  is to be determined by the working group for each supported OS.
+* Make the change as minimally invasive as possible to the image, runtime and distribution specs - that could be achieved over
+  delivering compatibility artifact type.
+* Deprecate os.version and os.features fields from the image spec that were mainly used for Windows.
+
+## Out of Scope
+
+* Change images build or container execution flow.
```

Comment:
This is a good rationale for the separated nature - we'd want the generation of this spec to be done separately from the standard build, etc. I haven't thought about the details but I suspect the extraction could be some kind of plugin or additional scan after those steps. I don't think we want them tangled, or to place additional requirements for this on existing building tools.

Reply:
We think to create a library (that could be used by different software like k8s operators) and a tool that could extract the spec or execute validation directly on the node. Most likely, we won't extend the already existing tools.